packages/block-editor/src/components/block-edit/edit.js

```diff
@@ -31,10 +31,19 @@ import BlockContext from '../block-context';
 const DEFAULT_BLOCK_CONTEXT = {};
 
 export const Edit = ( props ) => {
-	const { attributes = {}, name } = props;
+	const {
+		attributesContext = [],
+		setAttributes,
+		attributes = {},
+		name,
+	} = props;
 	const blockType = getBlockType( name );
 	const blockContext = useContext( BlockContext );
 
+	useEffect( () => {
+		attributesContext.map( ( ctxAttr ) => setAttributes( ctxAttr ) );
+	}, [ attributesContext ] );
```

Comment:
Could this be handled in the edit-post package? As in, we wouldn't pass attributesContext – we would just update the attributes through some store action.